Sources/XcodeGenKit/SchemeGenerator.swift

```diff
@@ -83,34 +83,56 @@ public class SchemeGenerator {
 
     public func generateScheme(_ scheme: Scheme) throws -> XCScheme {
 
-        func getBuildEntry(_ buildTarget: Scheme.BuildTarget) -> XCScheme.BuildAction.Entry {
+        func getBuildEntry(_ buildTarget: Scheme.BuildTarget) throws -> XCScheme.BuildAction.Entry {
+            let pbxProj: PBXProj
+            let projectFilePath: String
+            switch buildTarget.target.location {
+            case .project(let project):
+                guard let projectReference = self.project.getProjectReference(project) else {
+                    throw SchemeGenerationError.missingProject(project)
+                }
+                pbxProj = try XcodeProj(pathString: projectReference.path).pbxproj
+                projectFilePath = projectReference.path
+            case .local:
+                pbxProj = self.pbxProj
+                projectFilePath = "\(self.project.name).xcodeproj"
+            }
 
-            guard let pbxTarget = pbxProj.targets(named: buildTarget.target).first else {
-                fatalError("Unable to find target named \"\(buildTarget.target)\" in \"PBXProj.targets\"")
+            guard let pbxTarget = pbxProj.targets(named: buildTarget.target.name).first else {
+                throw SchemeGenerationError.missingTarget(buildTarget, projectPath: projectFilePath)
             }
 
-            guard let buildableName =
-                project.getTarget(buildTarget.target)?.filename ??
-                project.getAggregateTarget(buildTarget.target)?.name else {
-                fatalError("Unable to determinate \"buildableName\" for build target: \(buildTarget.target)")
+            let buildableName: String
+
+            switch buildTarget.target.location {
+            case .project:
+                buildableName = pbxTarget.productNameWithExtension() ?? pbxTarget.name
+            case .local:
+                guard let _buildableName =
+                        project.getTarget(buildTarget.target.name)?.filename ??
+                        project.getAggregateTarget(buildTarget.target.name)?.name else {
+                    throw SchemeGenerationError.unresolvableBuildableName(buildTarget)
```

Comment:
While there's nothing bad with having specific errors thrown here, checks like this occur in the SpecValidation phase. Adding a new set of SchemeGenerationErrors just introduces more maintenance. The rest of the codebase gracefully fails at points like this. Thoughts @kateinoigakukun, @giginet?

Reply:
You're right this condition will be not caused by user inputs, seems to be caused by an implementation error.
Other exceptions will be occurred by user inputs, however, here is not. 
It's better to call `fatalError` instead of throwing the exceptions here.

Reply:
Sorry @kateinoigakukun, could you revert to how you had it before then